.github/workflows/ci.yml

```diff
@@ -0,0 +1,141 @@
+
+      - name: Download and install toolchain
+        if: steps.cache-toolchain.outputs.cache-hit != 'true'
+        run: make arm_sdk_install
+
+  build:
+    name: Build
+    needs: setup
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        target: ${{ fromJson(needs.setup.outputs.targets) }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Fetch toolchain from cache
+        uses: actions/cache@v2
+        id: cache-toolchain
+        with:
+          path: tools/
+          key: ${{ runner.os }}-${{ hashFiles('make/tools.mk') }}
+
+      - name: Build target
+        run: make EXTRA_FLAGS=-Werror ${{ matrix.target }}
+
+      - name: Publish build artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: firmware
+          path: obj/*.hex
+
+  test:
+    name: Test
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install dependencies
+        run: sudo apt-get install -y libblocksruntime-dev
+
+      - name: Run sanity checks
+        run: make EXTRA_FLAGS=-Werror checks
+
+      - name: Run all unit tests
+        run: make EXTRA_FLAGS=-Werror test-all
+
+  release:
+    name: Release Nightly
+    if: ${{ github.event_name == 'push' }}
+    needs: [build, test]
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout nightly builds repository
+        uses: actions/checkout@v2
+        with:
+          repository: ${{ env.repoNightly }}
+
+      - name: Push empty tagged commit
+        run: |
+          git config user.name "github-actions"
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git commit --allow-empty -m "Release ${{ github.repository }} build ${{ github.sha }}"
+          git tag "${{ github.repository }}/${{ github.sha }}"
+          git push && git push --tags
+
+      - name: Fetch build artifacts
+        uses: actions/download-artifact@v2
+        with:
+          name: firmware
+          path: release/
+
+      - name: Release
+        uses: softprops/action-gh-release@1e07f4398721186383de40550babbdf2b84acfc5 # v0.1.14
+        with:
+          files: release/**
+          repository: ${{ env.repoNightly }}
+          name: '${{ github.repository }}: ${{ github.ref_name }}'
+          tag_name: ${{ github.repository }}/${{ github.sha }}
+          draft: false
+          prerelease: false
+          fail_on_unmatched_files: true
+          body: |
+            ${{ env.releaseNotes }}
+
+            ### Repository:
+            ${{ github.repository }} ([link](${{ github.event.repository.html_url }}))
+
+            ### Branch:
+            ${{ github.ref_name }} ([link](${{ github.event.repository.html_url }}/tree/${{ github.ref_name }}))
+
+            ### Latest changeset:
+            ${{ github.event.head_commit.id }} ([link](${{ github.event.head_commit.url }}))
+
+            ### Changes:
+            ${{ github.event.head_commit.message }}
```

Comment:
Release should be block on forks

Reply:
I don't think it's necessary to check if running on a fork since it only runs on master/maintenance branches which forks should not typically use. It's not possible for forks to release on the betaflight repos if that's what you mean.

Reply:
@mathiasvr it would be good to simply check that the release action only runs on the main repo, we will get complaints from people who have forks that end up with failures (we have had it with other repos).